### PR TITLE
New method: async version of sweeping a linestring to a mesh.

### PR DESCRIPTION
**Pull request by EarlinLutz**
(Request of Nicolas Woodward)

### CODE REVIEW

File: core/geometry/src/polyface/PolyfaceQuery.ts [hunk 1]
```diff
@@ -321,66 +318,65 @@ export class PolyfaceQuery {
...
+      const visitor = polyface.createVisitor(0);
+      for (visitor.reset(); visitor.moveToNextFacet();) {
+        context.projectToPolygon(visitor.point, announce, polyface, visitor.currentReadIndex());
+      }
+    }
+  }
+
+
+  /** Execute context.projectToPolygon until its work estimates accumulate to workLimit  */
+  private static async continueAnnouunceSweepLinestringToConvexPolyfaceXY(
```
Comment (mgooding) [inline core/geometry/src/polyface/PolyfaceQuery.ts:332]: typo, "Annouunce"

File: core/geometry/src/polyface/PolyfaceQuery.ts [hunk 2]
```diff
@@ -321,66 +318,65 @@ export class PolyfaceQuery {
...
+    context: SweepLineStringToFacetContext, visitor: PolyfaceVisitor, announce: AnnounceDrapePanel): Promise<number> {
+    let workCount = 0;
+    while ((workCount < this.asyncWorkLimit) && visitor.moveToNextFacet()) {
+      workCount += context.projectToPolygon(visitor.point, announce, visitor.clientPolyface()!, visitor.currentReadIndex());
```
Comment (mgooding) [inline core/geometry/src/polyface/PolyfaceQuery.ts:336]: Can this assert clientPolyface is defined instead of using !

**Review by mgooding — APPROVED** [review]

File: core/geometry/src/polyface/PolyfaceQuery.ts (hunk 2 above)
Comment (EarlinLutz) [reply core/geometry/src/polyface/PolyfaceQuery.ts:336]: hm...If you've gotten this far the bang is safe

File: core/geometry/src/polyface/PolyfaceQuery.ts (hunk 1 above)
Comment (EarlinLutz) [reply core/geometry/src/polyface/PolyfaceQuery.ts:332]: Ah, uuuuuuuuuuuu... it's private, so I'll fix on next PR.

**Comment by EarlinLutz** [comment]
@Mergifyio backport release/2.10.0

**Pull request merged**